### PR TITLE
Bugfix/handle hidapi exceptions in sendReport

### PR DESCRIPTION
**Pull request by dalethomas81**
Handle an exception generated by hidapi in the `sendReport` thread. This exception is thrown when no controller is connected. Since the exception is not handled, the thread ends, and the service needs to be restarted to recover. By handling these exceptions, the service becomes fault tolerant and is able to recover automatically.  

- Added a Try/Exception to handle an IOError and an AttributeError.
- Added a `Connected` property to indicate when the device is not connected.

### CODE REVIEW

File: pydualsense/pydualsense.py [hunk 1]
```diff
@@ -114,6 +114,7 @@ def init(self) -> None:
         self.battery = DSBattery()
         self.conType = self.determineConnectionType()  # determine USB or BT connection
         self.ds_thread = True
+        self.connected = True
```
Comment (flok) [inline pydualsense/pydualsense.py:117]: Is this variable really needed? you only set it 3 times but never compare any value to it?

File: pydualsense/pydualsense.py [hunk 2]
```diff
@@ -227,18 +228,26 @@ def setRightMotor(self, intensity: int) -> None:
...
+
+                # write the report to the device
+                self.writeReport(outReport)
+            except IOError:
+                self.connected = False
+                break
+                
+            except AttributeError:
+                self.connected = False
+                break
```
Comment (flok) [inline pydualsense/pydualsense.py:250]: In my understanding you could also just break when a exception is thrown and the class variable connected is not needed

**Comment by flok** [comment]
Thanks for the pull request, i added some remarks :)

**Comment by dalethomas81** [comment]
Hey @flok - sorry, I am not seeing the remarks. Where can I find them?

**Comment by dalethomas81** [comment]
Btw, this fixes #48.

**Comment by flok** [comment]
> Hey @flok - sorry, I am not seeing the remarks. Where can I find them?

You should see them in this thread at the top as a review from me

**Comment by dalethomas81** [comment]
I believe you may not have submitted the review request as there are none. Do you mind trying again or pasting your review here? Thank you for your patience. I am enjoying using this in my project. image

**Comment by flok** [comment]
> I believe you may not have submitted the review request as there are none. Do you mind trying again or pasting your review here? Thank you for your patience. I am enjoying using this in my project. 

You are right, my bad. Mostly working with gitlab now a days and forgot to press the button. You should be able to see it now

File: pydualsense/pydualsense.py (hunk 1 above)
Comment (dalethomas81) [reply pydualsense/pydualsense.py:117]: Hey @flok - this is intended to be used externally to indicated the controller is connected.

As an example, here is how I am using it in my project. https://github.com/dalethomas81/ArfBotOS/blob/main/Controller/DualSenseServer.py#L380

Is there another way to know if it is connected? I am open to alternative approaches.

**Comment by dalethomas81** [comment]
No worries! I did respond to your review. Can you let me know if that is satisfactory? Thank you.

File: pydualsense/pydualsense.py (hunk 1 above)
Comment (flok) [reply pydualsense/pydualsense.py:117]: No, it's alright

**Pull request merged**